app/app/services/provider_search_service.rb

```diff
@@ -6,23 +6,151 @@ def initialize(client_agency_id)
   end
 
   def search(query = "")
-    SUPPORTED_PROVIDERS.map do |provider|
-      case provider
-      when :pinwheel
-        PinwheelService.new(@client_agency_config.pinwheel_environment).fetch_items(q: query)["data"].map do |result|
-          ResponseObjects::SearchResult.from_pinwheel(result)
-        end
-      when :argyle
-        ArgyleService.new(@client_agency_config.argyle_environment).items(query)["results"].map do |result|
-          ResponseObjects::SearchResult.from_argyle(result)
-        end
+    results = []
+    if SUPPORTED_PROVIDERS.include?(:argyle)
+      results = ArgyleService.new(@client_agency_config.argyle_environment).items(query)["results"].map do |result|
+        ResponseObjects::SearchResult.from_argyle(result)
       end
-    end.flatten
+    end
+
+    if results.length == 0 && SUPPORTED_PROVIDERS.include?(:pinwheel)
+      results = (PinwheelService.new(@client_agency_config.pinwheel_environment).fetch_items(q: query)["data"].map do |result|
+        ResponseObjects::SearchResult.from_pinwheel(result)
+      end << results).flatten!
```

Comment:
The `end << results).flatten!` is a bit unintuitive/surprising. Is it even necessary given that results is going to be empty? 

Would this work?

```suggestion
      results = PinwheelService.new(@client_agency_config.pinwheel_environment).fetch_items(q: query)["data"].map do |result|
        ResponseObjects::SearchResult.from_pinwheel(result)
      end
```

Reply:
Updated